src/mlpack/methods/reinforcement_learning/environment/acrobot.hpp

```diff
@@ -85,19 +85,27 @@ class Acrobot
   /*
    * Implementation of action for Acrobot
    */
-  enum Action
+  class Action
   {
-    negativeTorque,
-    zeroTorque,
-    positiveTorque,
+   public:
+    enum _action
+    {
+      negativeTorque,
+      zeroTorque,
+      positiveTorque,
+    };
+    // To store the action.
+    Action::_action action;
```

Comment:
With the modification, you have to modify all constructs of the form:

https://github.com/mlpack/mlpack/blob/7adc6fb89ef8365763f7768c8691bfeacdc1b656/src/mlpack/methods/reinforcement_learning/worker/one_step_sarsa_worker.hpp#L255

as well, since `action` doesn't reference the `enum`, but `action.action` does.

Reply:
Yeah, I intend to do that in the next commit. I think I've used incorrect reference [here](https://github.com/mlpack/mlpack/pull/2497#issuecomment-653491524). Anyways, I'll make this change and get back with my question